string.rb

```diff
@@ -1,556 +1,556 @@
+#   h = {foo: 'bar', baz: 'bat'}
+#   'food'.sub('foo', h) # => "d"
 #
-#  <b>Block</b>
+# <b>Block</b>
 #
-#  In the block form, the current match string is passed to the block;
-#  the block's return value becomes the replacing string:
+# In the block form, the current match string is passed to the block;
+# the block's return value becomes the replacing string:
 #
-#    s = '@'
-#   '1234'.gsub(/\d/) {|match| s.succ! } # => "ABCD"
+#   s = '@'
+#   '1234'.gsub(/\d/) { |match| s.succ! } # => "ABCD"
 #
-#  Special match variables such as <tt>$1</tt>, <tt>$2</tt>, <tt>$`</tt>,
-#  <tt>$&</tt>, and <tt>$'</tt> are set appropriately.
+# Special match variables such as <tt>$1</tt>, <tt>$2</tt>, <tt>$`</tt>,
+# <tt>$&</tt>, and <tt>$'</tt> are set appropriately.
 #
-#  == Whitespace in Strings
+# == Whitespace in Strings
 #
-#  In class +String+, _whitespace_ is defined as a contiguous sequence of characters
-#  consisting of any mixture of the following:
+# In the class +String+, _whitespace_ is defined as a contiguous sequence of characters
+# consisting of any mixture of the following:
 #
-#  - NL (null): <tt>"\x00"</tt>, <tt>"\u0000"</tt>.
-#  - HT (horizontal tab): <tt>"\x09"</tt>, <tt>"\t"</tt>.
-#  - LF (line feed): <tt>"\x0a"</tt>, <tt>"\n"</tt>.
-#  - VT (vertical tab): <tt>"\x0b"</tt>, <tt>"\v"</tt>.
-#  - FF (form feed): <tt>"\x0c"</tt>, <tt>"\f"</tt>.
-#  - CR (carriage return): <tt>"\x0d"</tt>, <tt>"\r"</tt>.
-#  - SP (space): <tt>"\x20"</tt>, <tt>" "</tt>.
+# - NL (null): <tt>"\x00"</tt>, <tt>"\u0000"</tt>.
+# - HT (horizontal tab): <tt>"\x09"</tt>, <tt>"\t"</tt>.
+# - LF (line feed): <tt>"\x0a"</tt>, <tt>"\n"</tt>.
+# - VT (vertical tab): <tt>"\x0b"</tt>, <tt>"\v"</tt>.
+# - FF (form feed): <tt>"\x0c"</tt>, <tt>"\f"</tt>.
+# - CR (carriage return): <tt>"\x0d"</tt>, <tt>"\r"</tt>.
+# - SP (space): <tt>"\x20"</tt>, <tt>" "</tt>.
 #
 #
-#  Whitespace is relevant for these methods:
+# Whitespace is relevant for the following methods:
 #
-#  - #lstrip, #lstrip!: strip leading whitespace.
-#  - #rstrip, #rstrip!: strip trailing whitespace.
-#  - #strip, #strip!: strip leading and trailing whitespace.
+# - #lstrip, #lstrip!: Strip leading whitespace.
+# - #rstrip, #rstrip!: Strip trailing whitespace.
+# - #strip, #strip!: Strip leading and trailing whitespace.
 #
-#  == +String+ Slices
+# == +String+ Slices
 #
-#  A _slice_ of a string is a substring that is selected by certain criteria.
+# A _slice_ of a string is a substring selected by certain criteria.
 #
-#  These instance methods make use of slicing:
+# These instance methods utilize slicing:
 #
-#  - String#[] (aliased as String#slice): returns a slice copied from +self+.
-#  - String#[]=: returns a copy of +self+ with a slice replaced.
-#  - String#slice!: returns +self+ with a slice removed.
+# - String#[] (aliased as String#slice): Returns a slice copied from +self+.
+# - String#[]=: Mutates +self+ with the slice replaced and returns the given slice.
```

Comment:
```suggestion
# - String#[]=: Mutates +self+ with the slice replaced.
```

I don't think we should discuss return values of assignment methods, since when used in assignment, return values are ignored and the RHS is returned.